### PR TITLE
soft delete

### PR DESCRIPTION
**Pull request by LarryFinn**
this is a trait to mix into a dao class to enable soft delete easily.  Basically, it adds a default filter and then adds a function to softDelete.

### CODE REVIEW

File: src/scala/co/actioniq/luna/dao/SoftDeleteFilter.scala [hunk 1]
```diff
@@ -0,0 +1,49 @@
+package co.actioniq.luna.dao
+
+import slick.jdbc.JdbcProfile
+
+import scala.concurrent.Future
+
+trait SoftDeleteFilter[T <: DAOTable.Table[V, I, P], V <: IdModel[I], I <: IdType, P <: JdbcProfile, D] extends DefaultFilter[T, V, I, P] {
+  protected val profile: JdbcProfile
+  import profile.api._
+  def softNotDeletedFilter(t: T): Rep[Option[Boolean]]
```
Comment (alexmoore0) [inline src/scala/co/actioniq/luna/dao/SoftDeleteFilter.scala:10]: Why `Option` as the return type of the filter function?

File: test/scala/co/actioniq/luna/example/KittyDAO.scala [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+    original: Kitty
+  )(implicit ec: ExecutionContext): DBIOAction[FormValidatorMessageSeq, NoStream, Effect.Read] = {
+    DBIO.successful(FormValidatorMessageSeq())
+  }
+
+  override def softNotDeletedFilter(t: KittyTable) = t.deletedAt.isEmpty
+
+  override def softDeleteFilterField(t: KittyTable)(implicit fType: JdbcType[Long]) = t.deletedAt
+
+  override def softDeleteFilterUpdate() = Some(12345)
```
Comment (alexmoore0) [inline test/scala/co/actioniq/luna/example/KittyDAO.scala:49]: How would this work for updating a row if `deletedAt` is not on the `Kitty` model case class?

File: src/scala/co/actioniq/luna/dao/SoftDeleteFilter.scala (hunk 1 above)
Comment (LarryFinn) [reply src/scala/co/actioniq/luna/dao/SoftDeleteFilter.scala:10]: i think for some reason it handles both Rep[Boolean] and Rep[Option[Boolean]]

Comment (LarryFinn) [reply src/scala/co/actioniq/luna/dao/SoftDeleteFilter.scala:10]: maybe type erasure something?

File: test/scala/co/actioniq/luna/example/KittyDAO.scala (hunk 2 above)
Comment (LarryFinn) [reply test/scala/co/actioniq/luna/example/KittyDAO.scala:49]: i think it would be needed, not sure how else to implement it

**Pull request merged**